src/libraries/System.Linq/tests/IndexTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class IndexTests : EnumerableTests
+    {
+        [Fact]
+        public void Index_SourceIsNull_ArgumentNullExceptionThrown()
+        {
+            IEnumerable<int> source = null;
+            
+            AssertExtensions.Throws<ArgumentNullException>("source", () => source.Index());
+        }
+
+        [ConditionalFact(typeof(TestEnvironment), nameof(TestEnvironment.IsStressModeEnabled))]
+        public void IndexOverflows()
+        {
+            var infiniteWhere = new FastInfiniteEnumerator<int>().Index();
```

Comment:
Perhaps constrain down to the smallest possible enumerable that is known to cause this. You also avoid an infinite loop in the hypothetical case that we regress behavior.
```suggestion
            var infiniteWhere = new FastInfiniteEnumerator<int>().Take(int.MaxValue + 2).Index();
```

Reply:
Would it also be possible to add a test showing that this doesn't throw when you take `int.MaxValue + 1` elements?